### PR TITLE
Initial port of the infrastructure testing library from Cobalt

### PR DESCRIPTION
**Pull request by AdanOrtiz**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------

Issue Number: N/A

## What is the new behavior?
-------------------------------------

- This Go package offers abstractions over the popular [Terratest](https://github.com/gruntwork-io/terratest) library in order to abstract some common testing patterns that have been identified through multiple projects that deploy [Infrastructure as Code](https://docs.microsoft.com/en-us/azure/devops/learn/what-is-infrastructure-as-code) (*IAC*) using [Terraform](https://www.terraform.io/). 
- The abstractions offered here can be used along side existing Terratest code and are quite easy to drop into existing projects. 

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

**Review by nmiodice — COMMENTED** [review]
I noticed that the github action failed - https://github.com/microsoft/terratest-abstraction/pull/1/checks?check_run_id=592431955

I've submitted what I think should fix it.

Edit: woot! https://github.com/microsoft/terratest-abstraction/runs/592633490?check_suite_focus=true

**Comment by nmiodice** [comment]
Should we add a lint check to the CICD?

**Review by nmiodice — APPROVED** [review]
Ship It! May be worth to get another set of eyes here since we collaborated, but this LGTM.

```
              |    |    |
             )_)  )_)  )_)
            )___))___))___)\
           )____)____)_____)\\
         _____|____|____|____\\\__
---------\                   /---------
  ^^^^^ ^^^^^^^^^^^^^^^^^^^^^
    ^^^^      ^^^^     ^^^    ^^
         ^^^^      ^^^
```

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        fi
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
+
+    - name: Build Samples
+      run: (cd samples && go build -v ./...)
```
Comment (TechnicallyWilliams) [inline .github/workflows/go.yml:54]: Nit: new line is missing.

**Review by TechnicallyWilliams — APPROVED** [review]

**Pull request merged**